be/src/vec/exec/format/json/new_json_reader.cpp

```diff
@@ -1007,4 +1015,560 @@ Status NewJsonReader::_read_one_message(std::unique_ptr<uint8_t[]>* file_buf, si
+    // set
+    size_t cur_row_count = columns[0]->size();
+    bool has_valid_value = false;
+    // iterate through object, simdjson::ondemond will parsing on the fly
+    for (auto field : *value) {
+        std::string_view key = field.unescaped_key();
+        auto iter = _slot_desc_index.find(std::string(key));
+        if (iter == _slot_desc_index.end()) {
+            // This key is not exist in slot desc, just ignore
+            continue;
+        }
+        simdjson::ondemand::value val = field.value();
+        RETURN_IF_ERROR(_simdjson_write_data_to_column(val, slot_descs[iter->second],
+                                                       columns[iter->second].get(), valid));
+        if (!(*valid)) {
+            return Status::OK();
+        }
+        has_valid_value = true;
+    }
+    if (!has_valid_value) {
+        RETURN_IF_ERROR(
+                _append_error_msg(value, "All fields is null, this is a invalid row.", "", valid));
+        return Status::OK();
+    }
+
+    // fill missing slot
+    int nullcount = 0;
+    int ctx_idx = 0;
+    for (auto slot_desc : slot_descs) {
+        if (!slot_desc->is_materialized()) {
+            continue;
+        }
+        int dest_index = ctx_idx++;
+        auto* column_ptr = columns[dest_index].get();
+        if (column_ptr->size() < cur_row_count + 1) {
+            DCHECK(column_ptr->size() == cur_row_count);
+            column_ptr->assume_mutable()->insert_default();
+            ++nullcount;
+        }
+        DCHECK(column_ptr->size() == cur_row_count + 1);
+    }
+    // There is at least one valid value here
+    DCHECK(nullcount < columns.size());
+    *valid = true;
+    return Status::OK();
+}
+
+Status NewJsonReader::_simdjson_write_data_to_column(simdjson::ondemand::value& value,
+                                                     SlotDescriptor* slot_desc,
+                                                     vectorized::IColumn* column, bool* valid) {
+    // write
+    vectorized::ColumnNullable* nullable_column = nullptr;
+    vectorized::IColumn* column_ptr = nullptr;
+    if (slot_desc->is_nullable()) {
+        nullable_column = assert_cast<vectorized::ColumnNullable*>(column);
+        column_ptr = &nullable_column->get_nested_column();
+    }
+
+    // TODO: if the vexpr can support another 'slot_desc type' than 'TYPE_VARCHAR',
+    // we need use a function to support these types to insert data in columns.
+    ColumnString* column_string = assert_cast<ColumnString*>(column_ptr);
+    if (value.is_null()) {
+        if (column->is_nullable()) {
+            // insert_default already push 1 to null_map
+            nullable_column->insert_default();
+        } else {
+            RETURN_IF_ERROR(_append_error_msg(
+                    nullptr, "Json value is null, but the column `{}` is not nullable.",
+                    slot_desc->col_name(), valid));
+            return Status::OK();
```

Comment:
Why return OK here?

Reply:
_append_error_msg will append err msg and continue to process next line